dbms/src/Interpreters/Join.h

```diff
@@ -192,6 +204,28 @@ class Join
     };
 
 
+    struct ProbeProcessInfo
+    {
+        Block block;
+        UInt64 max_block_size;
+        size_t start_row;
+        size_t end_row;
+        bool all_rows_joined_finish;
+        bool block_full;
+
+        ProbeProcessInfo(Block block_, UInt64 max_block_size_, size_t start_row_, size_t end_row_, bool all_rows_joined_finish_, bool block_full_)
```

Comment:
It seems that providing a default constructor is ok, and don't need to `std::make_shared<ProbeProcessInfo>(Block{}, max_block_size, 0, 0, true, false)`

Reply:
> It seems that providing a default constructor is ok, and don't need to `std::make_shared<ProbeProcessInfo>(Block{}, max_block_size, 0, 0, true, false)`

max_block_size  need to init by constructor, changed.

Reply:
> It seems that providing a default constructor is ok, and don't need to `std::make_shared<ProbeProcessInfo>(Block{}, max_block_size, 0, 0, true, false)`

A needs to be initialized by constructor, changed.

Reply:
> It seems that providing a default constructor is ok, and don't need to `std::make_shared<ProbeProcessInfo>(Block{}, max_block_size, 0, 0, true, false)`

max_block_size needs to be initialized by constructor, changed